game/scripts/vscripts/items/manta.lua

```diff
@@ -0,0 +1,204 @@
+    -- Though the formation is always the same, the owner and the illusions take a random position in the formation and have all the same facing angle.
+      local position = GetImageLocation(origin, casterIndex, false, imageIndex, images_count)
+
+    -- Recasting this replaces the illusions from the previous cast which are currently under the owner's control.
+      if image ~= nil and IsValidEntity(image) then
+        if image:IsAlive() then
+          image:ForceKill(false)
+        end
+      end
+
+      --DebugDrawLine(origin, position, 255, 0, 0, true, 20)
+
+      image = CreateUnitByName(
+        caster:GetUnitName(), --szUnitName
+        position,             --vLocation
+        true,                 --bFindClearSpace
+        caster,               --hNPCOwner
+        nil,                  --hUnitOwner
+        teamID                --iTeamNumber
+      )
+
+      image:SetForwardVector(forwardVector)
+
+      image:SetPlayerID(playerID)
+      image:SetControllableByPlayer(playerID, true)
+
+      --Level up the image to the caster's level.
+      local level = caster:GetLevel()
+      for i = 1, level - 1 do
+        image:HeroLevelUp(false)
+      end
+
+      --Set the image's available skill points to 0 and teach it the abilities the caster has.
+      image:SetAbilityPoints(0)
+      for abilityIndex = 0, 15 do
+        local casterAbility = caster:GetAbilityByIndex(abilityIndex)
+        if casterAbility ~= nil then
+          local imageAbility = image:FindAbilityByName(casterAbility:GetAbilityName())
+          if imageAbility ~= nil then
+            imageAbility:SetLevel(casterAbility:GetLevel())
+          end
+        end
+      end
+
+      --Recreate the caster's items for the image.
+      for itemSlot = DOTA_ITEM_SLOT_1, DOTA_ITEM_SLOT_5 do
+        local casterItem = caster:GetItemInSlot(itemSlot)
+        if casterItem ~= nil then
+          local imageItem = CreateItem(casterItem:GetName(), image, image)
+          image:AddItem(imageItem)
+        end
+      end
+
+      -- Set Status
+      image:SetHealth(caster:GetHealth())
+      image:SetMana(caster:GetMana())
+
+      -- modifier_illusion controls many illusion properties like +Green damage not adding to the unit damage, not being able to cast spells and the team-only blue particle
+      image:AddNewModifier(caster, ability, "modifier_illusion", {
+        duration = ability:GetSpecialValueFor("illusion_duration"),
+        outgoing_damage = image_outgoing_damage,
+        incoming_damage = image_incoming_damage
+      })
+
+      image:MakeIllusion()  --Without MakeIllusion(), the unit counts as a hero, e.g. if it dies to neutrals it says killed by neutrals, it respawns, etc.  Without it, IsIllusion() returns false and IsRealHero() returns true.
+
+      ability.images[imageIndex] = image
+    end
+
+    -- Allow Render of Caster
+    caster:RemoveNoDraw()
+
+    ParticleManager:DestroyParticle(self.particle, false)
```

Comment:
Should also release index: `ParticleManager:ReleaseParticleIndex(self.particle)`